test/e2e/util_gatewayapi_test.go

```diff
@@ -271,6 +271,21 @@ func createGatewayClass(name, controllerName string) (*gatewayapiv1.GatewayClass
 	return gatewayClass, nil
 }
 
+// createCRD creates the CRD with the given name or retrieves it if already exists.
+func createCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error) {
+	crd := buildGWAPICRDFromName(name)
+	if err := kclient.Create(context.Background(), crd); err != nil {
+		if kerrors.IsAlreadyExists(err) {
+			if err := kclient.Get(context.Background(), types.NamespacedName{Name: name}, crd); err != nil {
+				return nil, fmt.Errorf("failed to get crd: %w", err)
+			}
+			return crd, nil
+		}
+		return nil, fmt.Errorf("failed to create crd: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to create crd %s: %w", name, err)
```

Reply:
I returned to [the previous idea of using `ensureExperimentalCRDs`](https://github.com/openshift/cluster-ingress-operator/pull/1200#discussion_r2005192321), so this function doesn't exist anymore.